src/ApplicationInsights.Kubernetes/Utilities/K8sDetectors/K8sDetector.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace Microsoft.ApplicationInsights.Kubernetes.Utilities
+{
+    /// <summary>
+    /// Detects if the application is running in containers managed by Kubernetes.
+    /// </summary>
+    public abstract class K8sDetector
+    {
+        K8sDetector _nextDetector;
+
+        public K8sDetector(K8sDetector nextDetector)
+        {
+            _nextDetector = nextDetector;
+        }
+
+        /// <summary>
+        /// Returns true when the application is running in containers managed by Kubernetes.
+        /// </summary>
+        /// <returns></returns>
+        public bool IsRunningInKubernetes()
```

Comment:
This implementation is never used (every concrete class overrides it). Would a simple interface `IK8sDetector` be sufficient?

Reply:
As a matter of fact, this is where chain logic being considered, so that all the derived classes only need to consider its very specific prediction.
For example, the linux detector only detects for linux; the windows detector only detects for windows.

Reply:
Off the topic, since you touched the topic on the interface, generally, I prefer to use an interface over an abstract class, simply because an interface is easier to mock. However, when there comes with shared logic, I think we shall not afraid of making use of it. What's your idea?